### PR TITLE
Increase meta-data fidelity

### LINKED ISSUE #70: Increase AMQP 091 -> AMQP 1.0 meta data fidelity
When forwarding a 091 message to 1.0 a lot of the meta-data included in the "properties" map is discarded and not included in the AMQP 1.0 data. Any properties that do not map directly to the amqp 1.0 properties should be added to the application properties instead. 

https://github.com/rabbitmq/rabbitmq-shovel/blob/499bc1c4e86c65535d0b34c7737ec4d3e1734f83/src/rabbit_amqp10_shovel.erl#L322-L324

### PR DESCRIPTION
**Pull request by kjnilsson**
When shovelling 0.9.1 messages to 1.0.

Map any basic properties that map directly to AMQP 1.0 properties and
add the remaining properties (including routing key) to application
properties if they have compatible types.
Fixes #70

### CODE REVIEW

File: src/rabbit_amqp10_shovel.erl [hunk 1]
```diff
@@ -353,21 +353,66 @@ add_forward_headers(#{dest := #{cached_forward_headers := Props}}, Msg) ->
...
+      end, Msg, Props).
 
 gen_unique_name(Pre0, Post0) ->
     Pre = rabbit_data_coercion:to_binary(Pre0),
     Post = rabbit_data_coercion:to_binary(Post0),
     Id = bin_to_hex(crypto:strong_rand_bytes(8)),
     <<Pre/binary, <<"_">>/binary, Id/binary, <<"_">>/binary, Post/binary>>.
 
+to_binary(X) ->
+    rabbit_data_coercion:to_binary(X).
```
Comment (michaelklishin) [inline src/rabbit_amqp10_shovel.erl:406]: Why not import this function?

**Comment by michaelklishin** [comment]
Backported to `v3.8.x`.

File: src/rabbit_amqp10_shovel.erl (hunk 1 above)
Comment (kjnilsson) [reply src/rabbit_amqp10_shovel.erl]: I never import as it doesn't work with my code navigation :)

**Comment by kjnilsson** [comment]
Cheers!

**Pull request merged**